x-pack/plugin/profiler/src/main/java/org/elasticsearch/xpack/profiler/ProfilingIndexManager.java

```diff
@@ -354,7 +354,14 @@ public ProfilingIndex withGeneration(String generation) {
         }
 
         public boolean isMatchWithoutVersion(String indexName) {
-            return indexName.startsWith("." + namePrefix);
+            String expectedPrefix = "." + namePrefix + "-v";
+            return indexName.startsWith(expectedPrefix) && isVersionNumber(indexName, expectedPrefix.length());
+        }
+
+        private boolean isVersionNumber(String name, int startIndex) {
+            final int versionNumberLength = 3;
+            String versionNumberCandidate = name.substring(startIndex, Math.min(startIndex + versionNumberLength, name.length()));
+            return versionNumberCandidate.length() == versionNumberLength && versionNumberCandidate.chars().allMatch(Character::isDigit);
```

Comment:
> Correct. This means that we check that each character is within that range of code points.

Sorry, I didn't write it explicitly: My point was, let's only check for ASCII digits ('0'..'9') to avoid any ambiguities.

> However, we'd still need to do another substring call in isVersionNumber so I think it's fine to do it in one go?

Sure, nevermind (I was just thinking about some simple code without another substring).

> Cluster state update handling should be lightweight

:+1:

Reply:
> My point was, let's only check for ASCII digits ('0'..'9') to avoid any ambiguities.

Done in https://github.com/elastic/elasticsearch/pull/97710/commits/80f52858fe51290465e54cefdb5dd260936a9d3c.